tuplex/codegen/src/FunctionRegistry.cc

```diff
@@ -1628,6 +1631,99 @@ namespace tuplex {
             return SerializableValue();
         }
 
+        SerializableValue FunctionRegistry::createIteratorRelatedSymbolCall(tuplex::codegen::LambdaFunctionBuilder &lfb,
+                                                                            llvm::IRBuilder<> &builder,
+                                                                            const std::string &symbol,
+                                                                            const python::Type &argsType,
+                                                                            const python::Type &retType,
+                                                                            const std::vector<tuplex::codegen::SerializableValue> &args,
+                                                                            IteratorInfo *iteratorInfo) {
+            if(symbol == "iter") {
+                return createIterCall(lfb, builder, argsType, retType, args);
+            }
+
+            if(symbol == "zip") {
+                return createZipCall(lfb, builder, argsType, retType, args, iteratorInfo);
+            }
+
+            if(symbol == "enumerate") {
+                return createEnumerateCall(lfb, builder, argsType, retType, args, iteratorInfo);
+            }
+
+            if(symbol == "next") {
+                return createNextCall(lfb, builder, argsType, retType, args, iteratorInfo);
+            }
+
+            return SerializableValue(nullptr, nullptr);
```

Comment:
maybe add a warning here? because what if an unsupported symbol is requested?

Reply:
Add error message.